client/components/payment-activity/payment-data-tile.tsx

```diff
@@ -24,7 +24,7 @@ interface PaymentDataTileProps {
 	/**
 	 * The currency code for the amount displayed.
 	 */
-	currencyCode: string;
+	currencyCode?: string;
```

Comment:
I wonder why this is optional. Actually, I also wonder why the other properties, amount, tooltip, etc, are optional.

Is it because when data is still being fetched or when an error returned by server, `currencyCode` and the other will be undefined?

This makes me think of error handling. When server returns an error, I think we should not render the payment activity card altogether.

What do you think? I think that work should be done in a separate PR though. I can take on it.

Reply:
Yes, due to the way the components are currently rendered, `<Loadable>` is rendered if passed `undefined` values.

In this PR, I may look into making this more explicit for the loading case by using discriminate types, e.g.
- interface A: props to expect when loading.
- interface B: props to expect when loading is complete.

> When server returns an error, I think we should not render the payment activity card altogether.

But yes, I agree the error UI state might be a new new issue / PR 💯